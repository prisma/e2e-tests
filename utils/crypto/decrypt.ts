#!/usr/bin/env ts-node

import * as childProcess from 'child_process'
import util from 'util'

const exec = util.promisify(childProcess.exec)

const PASS = process.env.AES_256_PASS

/**
 * Decrypt a value with AES-256
 * @param value 
 * @returns 
 */
export async function decrypt(value: string) {
    const input = `echo "${value}"`
    const decrypt = `openssl aes-256-cbc -d -a -pass pass:${PASS} -pbkdf2`
    const output = await exec(`${input} | ${decrypt}`)

    if (output.stderr) process.exit(1)

    return output.stdout.trim()
}

/**
 * Decrypt a value via the command line
 * @returns 
 */
async function main() {
    if (require.main !== module) return

    console.log(await decrypt(process.argv[2]))
}

main()