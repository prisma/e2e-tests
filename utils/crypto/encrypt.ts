#!/usr/bin/env ts-node

import * as childProcess from 'child_process'
import util from 'util'

const exec = util.promisify(childProcess.exec)

const PASS = process.env.AES_256_PASS

/**
 * Encrypt a value with AES-256
 * @param value 
 * @returns 
 */
export async function encrypt(value: string) {
    const input = `echo "${value}"`
    const encrypt = `openssl aes-256-cbc -a -salt -pass pass:${PASS} -pbkdf2`
    const output = await exec(`${input} | ${encrypt}`)

    if (output.stderr) process.exit(1)

    return output.stdout.trim().replace(/\n/g, '')
}

/**
 * Encrypt a value via the command line
 * @returns 
 */
async function main() {
    if (require.main !== module) return

    console.log(await encrypt(process.argv[2]))
}

main()