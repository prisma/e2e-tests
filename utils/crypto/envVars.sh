#!/usr/bin/env bash

# equivalent of dirname in node.js
DIR=$(dirname "${BASH_SOURCE[0]}")

# execute what ts-node has printed
eval $(yarn --cwd $DIR ts-node $DIR/envVars.ts "$@")
