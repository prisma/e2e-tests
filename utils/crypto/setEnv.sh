#!/usr/bin/env bash

# disable cmd printings in this script
# we don't want sensitive info to leak
if [[ $- == *x* ]]; then # if -x is on
    set +x # then we disable it
    HAS_X_FLAG=true
fi

# equivalent of a __dirname in node.js
DIR=$(dirname "${BASH_SOURCE[0]}")

# makes sure which node_modules to use
MOD_DIR="$DIR/../../node_modules"

# workaround for an error with sourcing
TS_NODE=$(yarn --modules-folder $MOD_DIR bin ts-node)

# execute what ts-node has printed out
eval $($TS_NODE $DIR/setEnv.ts "$@")

# reactivate -x flag for cmd printing
if [[ "${HAS_X_FLAG-}" == true ]]; then
    set -x
fi
