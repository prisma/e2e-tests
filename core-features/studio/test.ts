import fs from 'fs'
import { createHash } from 'crypto'
import fetch from 'node-fetch'

const STUDIO_PORT = 5555
const SCHEMA_HASH = createHash('md5')
  .update(fs.readFileSync('./prisma/schema.prisma'))
  .digest('hex')

describe('Studio', () => {
  test('can load up the frontend correctly', async () => {
    let res = await fetch(`http://localhost:${STUDIO_PORT}`)
    expect(res.status).toBe(200)

    res = await fetch(`http://localhost:${STUDIO_PORT}/index.js`)
    expect(res.status).toBe(200)

    res = await fetch(`http://localhost:${STUDIO_PORT}/index.css`)
    expect(res.status).toBe(200)
  })

  test('can make queries', async () => {
    const res = await fetch(`http://localhost:${STUDIO_PORT}/api`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        requestId: 1,
        channel: 'prisma',
        action: 'clientRequest',
        payload: {
          data: {
            schemaHash: SCHEMA_HASH,
            query: 'prisma.user.findMany()',
          },
        },
      }),
    }).then((r) => r.json())

    expect(res).toMatchSnapshot()
  })
})
