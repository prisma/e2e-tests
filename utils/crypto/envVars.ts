#!/usr/bin/env ts-node

import { decrypt } from "./decrypt";

const envVars = {
    CF_DATA_PROXY_URL: "U2FsdGVkX1+lmUv1EuHjvyIv1lhPgqgIFnx7IkabaGkchigC2NHjZ6FVIBTC0pMNNo1OvdqLB6vGqHf7CFA+bXgtqkSdEStLQyys1L6WQnEUNEJn6PvtDwT5DOXE0gYHoUitFM9VP9yQI8BolMGZWYWpV7PT8UTxZuf7o+EaTqY8AsXXRF6YrKQGdM1aviY6",
    CF_ACCOUNT_ID: "U2FsdGVkX1/kgKmF6Nc3rIIkben8az6boItdyzPJFj5vEPoxnWtfpyjEO8E6voHLB4Zl/+kddgTA14uboA+40A==",
    CF_API_TOKEN: "U2FsdGVkX18fx8PpClVvhLV9TuBGm//20tdPUYRowj/hBUw+r7nECbfZ7DBPKgI9lelSXXKOXD4FFNg6FPpURA==",

    VERCEL_DATA_PROXY_URL: "U2FsdGVkX1+lmUv1EuHjvyIv1lhPgqgIFnx7IkabaGkchigC2NHjZ6FVIBTC0pMNNo1OvdqLB6vGqHf7CFA+bXgtqkSdEStLQyys1L6WQnEUNEJn6PvtDwT5DOXE0gYHoUitFM9VP9yQI8BolMGZWYWpV7PT8UTxZuf7o+EaTqY8AsXXRF6YrKQGdM1aviY6",
    VERCEL_EDGE_FUNCTIONS_ORG_ID: "U2FsdGVkX1+OynKKazwnZh8yRbrGoapHZmyTDNe2ryNIFCGS/x8ALH8xROfzCdz+",
    VERCEL_EDGE_FUNCTIONS_PROJECT_ID: "U2FsdGVkX1/rvuHFq0MrRcXU3LmvnzZQp8A9/WeoqYg8IllvAJhujSnQOVfSvbqvfBGzk7t/ewBV/uXyQ99Jwg=="
}

/**
 * Decrypt a given encrypted entry
 * @param name 
 * @returns 
 */
export function get(name: string) {
    if (envVars[name] === undefined) {
        console.error(`${name} does not exist`)
        process.exit(1)
    }

    return decrypt(envVars[name])
}

/**
 * Decrypt, export and mask env vars
 * @returns 
 */
async function main() {
    if (require.main !== module) return

    const envVarNames = process.argv.slice(2)
    for (const name of envVarNames) {
        console.log(`export ${name}="${await get(name)}";`)
        console.log(`echo "::add-mask::$${name}";`)
    }
}

main()