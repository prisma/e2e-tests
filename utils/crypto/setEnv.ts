#!/usr/bin/env ts-node

import { decrypt } from './decrypt';
import { envVars } from './envVars'

/**
 * Decrypt a given encrypted entry of `envVars`
 * @param name 
 * @returns 
 */
function get(name: string) {
    if (envVars[name] === undefined) {
        console.error(`${name} does not exist`)
        process.exit(1)
    }

    return decrypt(envVars[name])
}

/**
 * Decrypt and generate export & mask env vars shell commands
 * @returns 
 */
async function main() {
    if (require.main !== module) return

    const envVarNames = process.argv.slice(2)
    for (const name of envVarNames) {
        console.log(`export ${name}="${await get(name)}";`)
        console.log(`echo "::add-mask::$${name}";`)
    }
}

main()