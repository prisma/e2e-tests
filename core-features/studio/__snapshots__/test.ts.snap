// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Studio can make queries 1`] = `
Object {
  "action": "clientRequest",
  "channel": "-prisma",
  "payload": Object {
    "data": Object {
      "params": Object {
        "action": "findMany",
        "clientMethod": "user.findMany",
        "dataPath": Array [],
        "model": "User",
        "runInTransaction": false,
      },
      "response": Array [],
    },
    "error": null,
  },
  "requestId": 1,
}
`;
